macaroon.js

```diff
@@ -315,6 +337,94 @@ const Macaroon = class Macaroon {
   }
 
   /**
+    Returns the V2 JSON serialization of this macaroon.
+    @return {Object} JSON serialization of this macaroon.
+  */
+  serializeJson() {
+    const obj = {
+      v: 2, // version
```

Comment:
If the original `exportAsJSONObject` function was creating v1 objects it might be best to just keep that method for now, maybe mark it as deprecated, and then let the new method just produce v2 objects

Reply:
For backward compatibility with old clients, we'll need to keep the capability to serialise to v1 objects for a while, and needing to know which method to call depending on the client version seems a bit awkward. Hopefully the new "export" method which serialises according to the version that the macaroon was created with or deserialised from should suffice - we've found that approach has worked pretty well in the Go library.